examples/cord_epsim/main.c

```diff
@@ -64,25 +65,49 @@ static gcoap_listener_t listener = {
 
 int main(void)
 {
-    puts("CoAP simplified RD registration example!\n");
+    puts("Simplified CoRE RD registration example\n");
 
     /* fill riot info */
     sprintf(riot_info, "{\"ep\":\"%s\",\"lt\":%u}",
             cord_common_get_ep(), CORD_LT);
 
+    /* parse RD address information */
+    sock_udp_ep_t rd_ep = {
+        .family    = AF_INET6,
+        .netif     = SOCK_ADDR_ANY_NETIF,
+        .port      = RD_PORT,
+    };
+
+
+    /* parse RD server address */
+    if (ipv6_addr_from_str((ipv6_addr_t *)&rd_ep.addr.ipv6, RD_ADDR) == NULL) {
+        puts("error: unable to parse RD address from RD_ADDR variable");
+        return 1;
+    }
+
     /* register resource handlers with gcoap */
     gcoap_register_listener(&listener);
 
     /* print RD client information */
-    puts("RD client information:");
-    printf(" RD addr: %s\n", CORD_SERVER_ADDR);
-    printf(" RD port: %u\n", (unsigned)CORD_SERVER_PORT);
-    printf("      ep: %s\n", cord_common_get_ep());
-    printf("      lt: %is\n", (int)CORD_LT);
-
-    /* run the shell */
-    char line_buf[SHELL_DEFAULT_BUFSIZE];
-    shell_run(NULL, line_buf, SHELL_DEFAULT_BUFSIZE);
+    puts("epsim configuration:");
+    printf("         ep: %s\n", cord_common_get_ep());
+    printf("         lt: %is\n", (int)CORD_LT);
+    printf(" RD address: [%s]:%u\n\n", RD_ADDR, (unsigned)RD_PORT);
+
+    xtimer_sleep(STARTUP_DELAY);
+
+    while (1) {
+        printf("updating registration with RD [%s]:%u\n", RD_ADDR,
+               (unsigned)RD_PORT);
+        int res = cord_epsim_register(&rd_ep);
+        if (res == CORD_EPSIM_BUSY) {
```

Comment:
If registration is changed to return *busy* as the expected response, then this handling will need to change as well.

Reply:
did not change it :-) So no need to tough this line, right?

Reply:
Yes, looks good.